File: BOJ_백준/13417 카드 문자열 (덱).cpp
#include <iostream>
#include <deque>

using namespace std;

int main() {
    int T, N;
    cin >> T;

    while(T--) {
        cin >> N;
        deque <char> card(N);
        deque <char> que;
        for (auto& i : card)
            cin >> i;

        que.push_front(card.front());
        card.pop_front();

        while (!card.empty()) {
            char c = card.front();
            card.pop_front();
            if (c - 'A' <= que.front() - 'A')
                que.push_front(c);
            else
                que.push_back(c);
        }

        for (char i : que)
            cout << i;
        cout << "\n";
    }

    return 0;
}

File: BOJ_백준/13417 카드 문자열 (덱).py
import sys
from collections import deque

input = sys.stdin.readline

for _ in range(int(input())):
    N = int(input())
    card = deque(input().split())
    que = deque(card.popleft())
    
    while card :
        c = card.popleft()
        if c <= que[0] :
            que.appendleft(c)
        else :
            que.append(c)
    print(''.join(que))
